rapid7/metasploit-framework
# Add SMB Shadow Module: Direct SMB Session Takeover

## This module intercepts direct SMB connections on the LAN
Both the SMB Server and Client must be on the LAN.
The SMB Client must be authenticating to the Server as an Administrator.
This module is dependent on an external ARP spoofer.
The builtin ARP spoofer was not providing sufficient host discovery.
[Bettercap v1.6.2](https://github.com/picatz/bettercap) was used during the development of this module.

## Verification Steps

1. Ensure Windows SMB Client and Server are on the LAN
2. Run bettercap targeting both the SMB Client and Server
3. Start msfconsole
4. Do `use exploit/windows/smb/smb_shadow`
5. Do `run`
6. Make the SMB Client connect to the SMB Server as an Administrator
7. Receive a Meterpreter Session as SYSTEM on the SMB Server host

See the paper on the SMB Shadow Attack:
https://strontium.io/blog/introducing-windows-10-smb-shadow-attack

SideNote -- The [guide for accepting modules](https://github.com/rapid7/metasploit-framework/wiki/Guidelines-for-Accepting-Modules-and-Enhancements#module-additions) mentions the [BailiWicked module](https://www.rapid7.com/db/modules/auxiliary/spoof/dns/bailiwicked_domain/), which has a spelling error. (Baliwicked)
